src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -44,6 +44,43 @@ using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::OnOff;
 using chip::Protocols::InteractionModel::Status;
 
+namespace {
+
+/**
+ * For all ModeBase alias clusters on the given endpoint, if the OnOff feature is supported and
+ * the OnMode attribute is set, update the CurrentMode attribute value to the OnMode value.
+ * @param endpoint
+ */
+void UpdateModeBaseCurrentModeToOnMode(EndpointId endpoint)
+{
+    for (const auto & element : ModeBase::GetModeBaseInstances())
+    {
+        auto modeBaseAlias = element.second;
+        if (modeBaseAlias->GetEndpointId() == endpoint)
+        {
+            if (modeBaseAlias->HasFeature(ModeBase::Feature::kOnOff))
+            {
+                ModeBase::Instance * modeBaseInstance               = element.second;
+                ModeBase::Attributes::OnMode::TypeInfo::Type onMode = modeBaseInstance->GetOnMode();
+                if (!onMode.IsNull())
+                {
+                    Status status = modeBaseInstance->UpdateCurrentMode(onMode.Value());
+                    if (status == Status::Success)
+                    {
+                        ChipLogProgress(Zcl, "Changed the Current Mode to %x", onMode.Value());
+                    }
+                    else
+                    {
+                        ChipLogError(Zcl, "Failed to Changed the Current Mode to %x: %u", onMode.Value(), to_underlying(status));
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Suggest wrapping the body of the function with `#ifdef EMBER_AF_PLUGIN_MODE_BASE`

Reply:
Done.